portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,42 +43,63 @@
 @WebServlet("/form-content")
 public class DataServlet extends HttpServlet {
 
+private static class Message {
+    private String uniqueKey;
+    private String message;
+    //private String imageUrl;
+    private long timestamp;
+
+    public Message(Key uniqueKey, String message, long time) {
+      this.uniqueKey = KeyFactory.keyToString(uniqueKey);
```

Comment:
Suggestion: The purpose for Message class is to make conversion to JSON supportable (for changing field names and adding fields). Translating from Datastore entity key to string is unrelated to this purpose. I think "KeyFactory.keyToString(uniqueKey)" belongs to the place where Message constructor is called.